harvard-lts/mirador-analytics-plugin
# Updated Mirador Analytics Plugin

**Updated Mirador Analytics Plugin**

---

**JIRA Ticket**: [LTSVIEWER-53](https://jira.huit.harvard.edu/browse/LTSVIEWER-53)

- Other Relevant Links: [PR for adding plugin to Viewer](https://github.com/harvard-lts/mps-viewer/pull/52)

# What does this Pull Request do?

Following the [suggested workflow for mirador plugins](https://github.com/harvard-lts/mps-viewer/blob/main/docs/developing-mirador-plugins.md#suggested-workflow-for-mirador-plugins), this pull request updates the analytics plugin so that it can be reusable to others in the Mirador community using Google Tag Manager for their analytics

# How should this be tested?

- Check out this branch
- Run `npm install` to update the dependencies
- Run `npm run serve` to view the plugin in a local version of Mirador
- If you inspect the viewer, you should see analytics code inserted into two areas of the page, at the top of the `<head>` and the top of the `<body>`
- Check that the container id in both places is `GTM-XXXXXXX`
- If you have access to Harvard Library's account in Google Tag Manager, you can also change the container id to `GTM-WSHSDQ5` in the demoEntry.js file and preview the local environment to ensure that things are firing correctly ([Video for instructions: Check that analytics is firing in Curiosity QA](https://drive.google.com/file/d/1wB3iJphfdrj28KLRgw456cnKeMNcqKb2/view?usp=sharing))
- Documentation has also been added

Note: I started with a copy of the pdiiif plugin, editing files and paring down the dependencies as I saw fit. It would be good to glance over these to make sure there aren't any unnecessary dependencies left or that I haven't removed something that's important to its integration with the viewer

# Test coverage

Yes/No: Are changes in this pull-request covered by:

- unit tests? Np
- integration tests? No

# Interested parties

@enriquediaz
